CRM/Contribute/Form/ContributionBase.php

```diff
@@ -293,7 +293,7 @@ public function preProcess() {
       $isPayLater = CRM_Utils_Array::value('is_pay_later', $this->_values);
 
       if ($isMonetary &&
-        (!$isPayLater || !empty($this->_values['payment_processor']))
+        (!$isPayLater && !empty($this->_values['payment_processor']))
```

Comment:
Actually we have started to treat $payLater as a payment processor (the Manual class) in some flows - so I would want to be very careful about this change